observability/test/run_o11y_tests.py

```diff
@@ -0,0 +1,1174 @@
+        server_proc = subprocess.Popen([server_start_cmd] + server_args.split(),
+                                       stdout=self.sponge_log_out,
+                                       stderr=self.sponge_log_out,
+                                       env=env)
+        return server_proc
+
+    def start_client_in_subprocess(self,
+                                   action: str,
+                                   env: Optional[Dict[str, Any]] = None) -> subprocess.Popen:
+        client_action_cmd = os.path.join(self.get_repo_path(self.args.client_lang.toEnum()),
+                                         self.get_client_action_cmd(self.args.client_lang))
+        client_args = self.get_client_action_args(
+            self.args.client_lang,
+            ClientActionArgs(server_host = 'localhost',
+                             server_port = self.args.port,
+                             action = action,
+                             exporter_interval = WAIT_SECS_EXPORTER_FLUSH))
+        logger.info('Running %s' % action)
+        client_proc = subprocess.Popen([client_action_cmd] + client_args.split(),
+                                       stdout=self.sponge_log_out,
+                                       stderr=self.sponge_log_out,
+                                       env=env)
+        return client_proc
+
+    def setup_and_run_rpc_gke(self, action: str) -> None:
+        gke_env = GKETestEnv(self.args)
+        gke_env.load_kubernetes_context()
+        try:
+            gke_env.deploy_gke_resources()
+            server_start_cmd = './%s %s &' % (
+                self.get_server_start_cmd(self.args.server_lang),
+                self.get_server_start_args(self.args.server_lang,
+                                           ServerStartArgs(port = self.args.port))
+            )
+            gke_env.start_testservice_server(self.server_config, WAIT_SECS_SERVER_START,
+                                             server_start_cmd)
+            client_action_cmd = './%s %s' % (
+                self.get_client_action_cmd(self.args.client_lang),
+                self.get_client_action_args(
+                    self.args.client_lang,
+                    ClientActionArgs(server_host = gke_env.server_pod_ip,
+                                     server_port = self.args.port,
+                                     action = action,
+                                     exporter_interval = WAIT_SECS_EXPORTER_FLUSH)))
+            gke_env.run_client_commands(self.client_config, WAIT_SECS_CLIENT_ACTION, [
+                client_action_cmd,
+            ])
+            gke_env.kill_testservice_server()
+            self.wait_before_querying_o11y_data()
+        except Exception:
+            self.test_runner.set_exit_status(1)
+            logger.error(traceback.format_exc())
+        finally:
+            gke_env.clean_up_gke_resources()
+            self.sponge_log_out.close()
+
+    def setup_and_run_rpc_vm(self,
```

Comment:
Why the `_vm` part of the method name? I don't see a VM being set up here.

Reply:
Fixed. Now it's just `setup_and_run_rpc`.